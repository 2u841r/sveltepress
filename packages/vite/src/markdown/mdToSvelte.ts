import { compile } from 'mdsvex'
import type { MdsvexOptions } from 'mdsvex'
import LRUCache from 'lru-cache'
import type { Highlighter } from '../types.js'
const cache = new LRUCache<string, any>({ max: 1024 })

export default async ({ mdContent, filename, mdsvexOptions }: {
  mdContent: string
  filename: string
  highlighter?: Highlighter
  mdsvexOptions?: MdsvexOptions
}) => {
  const cacheKey = JSON.stringify({ filename, mdContent })
  const cached = cache.get(cacheKey)
  if (cached)
    return cached
  const transformedSvelteCode = await compile(mdContent, {
    extensions: ['.md'],
    filename,
    ...mdsvexOptions,
  }) || { code: '', data: {} }
  cache.set(cacheKey, transformedSvelteCode)

  return transformedSvelteCode
}
