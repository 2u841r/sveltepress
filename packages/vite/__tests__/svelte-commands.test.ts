import { describe, expect, it } from 'vitest'
import mdToSvelte from '../src/markdown/md-to-svelte'

describe('svelte built-in commands', async () => {
  it('commands', async () => {
    const template = `<script>
  const items = ['foo', 'bar', 'zoo']
  const boolValue = false
</script>
  
<ul>

{#each items as item, i}
{@const str = \`\${i + 1}:\${item}\`}
<li>
{item}
</li>
  
{/each}

</ul>

{#if boolValue}
  <h3>
    True
  </h3>
{:else if items}
  <h3>
    items
  </h3>
{:else}
  <h3>
    False
  </h3>
{/if}

{#await Promise.resolve(1)}
  <h3>
    Loading
  </h3>
{:then val}
  <h3>
    {val}
  </h3>
{:catch}
  <h3>
    Error
  </h3>
{/await}

{@debug items}

{@html "<div class="foo">Foo</div>"}
`
    const res = await mdToSvelte({
      mdContent: template,
      filename: 'svelte-commands.md',
    })
    expect(res.code).toMatchSnapshot()
  })
})
