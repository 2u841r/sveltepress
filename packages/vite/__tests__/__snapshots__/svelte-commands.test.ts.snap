// Vitest Snapshot v1

exports[`svelte built-in commands > commands 1`] = `
"<script>
  const items = ['foo', 'bar', 'zoo']
  const boolValue = false
</script>
<ul>
{#each items as item, i}
{@const str = \`\${i + 1}:\${item}\`}
<li>
{item}
</li>
{/each}
</ul>
{#if boolValue}
  <h3>
    True
  </h3>
{:else if items}
  <h3>
    items
  </h3>
{:else}
  <h3>
    False
  </h3>
{/if}
{#await Promise.resolve(1)}
  <h3>
    Loading
  </h3>
{:then val}
  <h3>
    {val}
  </h3>
{:catch}
  <h3>
    Error
  </h3>
{/await}
{@debug items}
{@html \\"<div class=\\"foo\\">Foo</div>\\"}"
`;
